blockchain/client.go

```diff
@@ -129,80 +135,74 @@ func (c *Client) ListenEvents(
 
 	// Sequence historical and live headers.
 	headersC := make(chan types.Header, 2)
-	g.Go(func() error {
-		defer close(headersC)
-
-		for header := range histHeadersC {
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case headersC <- header:
-			}
-		}
+	defer close(headersC)
 
-		for header := range liveHeadersC {
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case headersC <- header:
-			}
+	g.Go(func() error {
+		if err = forwardHeaders(ctx, histHeadersC, headersC); err != nil {
+			return err
 		}
 
-		return nil
+		return forwardHeaders(ctx, histHeadersC, headersC)
```

Comment:
Can we keep only one `forwardHeaders` invocation here?

Reply:
we need to invocations I fixed the other one.